### ISSUE TITLE
`constructor` can't be the name of class accessors and generators.

### ISSUE THREAD

**Issue by g-plane** [microsoft/TypeScript#48948]
# Bug Report

In classes, accessors (aka getters/setters) and generators can't be named with `constructor`. They're *Syntax Errors* defined in spec.

References:

- [Early errors of class definitions](https://tc39.es/ecma262/multipage/ecmascript-language-functions-and-classes.html#sec-class-definitions-static-semantics-early-errors). Please focus on the second point of the `MethodDefinition` section.
- Class accessors and generators are [SpecialMethod](https://tc39.es/ecma262/multipage/ecmascript-language-functions-and-classes.html#sec-static-semantics-specialmethod)s.

### 🔎 Search Terms

constructor
getter
setter
accessor
generator

### 🕗 Version & Regression Information

Tested on 4.5.5, 4.6.2 and Nightly (4.7).

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?target=99&ts=4.7.0-dev.20220504#code/MYGwhgzhAEDCCM0DeAoa7oHMCmAXawA9gHYS4BOArsLoeQBQCUyaGb5el5xr6Avr2gQ8BEmSo069AG5gQlbMyQCBKUJBiwATCzYAqIqQrVaDJSqA)

### 💻 Code

```ts
class C1 {
    get constructor() {
        return
    }
    set constructor(value) {}
}

class C2 {
    *constructor() {}
}
```

### 🙁 Actual behavior

No errors reported.

### 🙂 Expected behavior

TypeScript should report errors for misusing `constructor` as accessor name for Line 2 and Line 5, and report error for misusing `constructor` as generator for Line 9 in the example above.
